

**The *Stephanorhinus* and *Coelodonta* phylogenetic conundrum**

The phylogenetic relationship of *Stephanorhinus* and *Coelodonta* has been extensively studied over the last forty years (reviewed in Cappellini et al., 2019). Both morpho-anatomical and molecular data have been used to argue that *Stephanorhinus* represents a paraphyletic group that is closely related to Coelodonta. Mitochondrial genome-based phylogenetic studies (Kirillova et al., 2017; Margaryan et al., 2020) using a single representative per taxon have placed *Stephanorhinus kirchbergensis* as sister to *Coelodonta*, forming a clade with *Dicerorhinus sumatrensis* (Sumatran rhinoceros) as outgroup. Proteome sequencing data derived from an Early Pleistocene dental enamel from a *Stephanorhinus* tooth was used to argue that *Coelodonta* evolved from an early *Stephanorhinus* lineage, and that the *Stephanorhinus* genus is therefore paraphyletic, which was corroborated by Antoine et al ‘s study that utilised 278 morphological characters (Antoine et al., 2021). However, the close phylogenetic affinities between the *Stephanorhinus-Coelodonta* clade and *Dicerorhinus* recovered by all molecular analyses are at odds with morphology, as analyses of the latter consistently argue that *Stephanorhinus* and/or *Coelodonta* are constantly more closely related to the African rhinoceroses (e.g., Antoine et al., 2021). Thus, the cladogram differences between the morphological and molecular datasets could be attributed to either morphological homoplasy, or long-branch attraction, the latter of which could happen as only limited numbers of genes were analysed, and as many Miocene-Pliocene rhinocerotines cannot be included in molecular trees.


**Genome synteny and collinearity**

By conducting pairwise genome alignment between each de novo assembled rhinoceros genome and that of horse (Equus caballus), we did not detect any major within-scaffold segment reshuffling between the rhinoceros species, as well as for that between rhinoceroses and the horse. We then applied Circos (Krzywinski et al., 2009) to visualise the genome synteny and collinearity (Figure 1). The good within-scaffold collinearity between rhinoceros species to some extent guarantees the reliability and accuracy of the reference-based analyses for those historic and ancient samples.


![Figure 1](https://github.com/liushanlin/rhinoceros-comparative-genome/blob/main/additional%20resources/Equu_chr1.png)

**Figure 1. A demonstration example shows the pairwise genome alignments between the de novo-assembled genomes and horse using chromosome #1**. Species can be identified by the first letter of each scaffold ID: “w” represents white rhinoceros (red lines), “b” represents black rhinoceros (blue lines), “s” represents Sumatran rhinoceros (green lines) and “i” represents greater one-horned rhinoceros (orange lines). The figure only shows scaffolds of length > 5 Mb and alignment blocks of length > 100 kb.

> **Reference**
> 
> Krzywinski, M., Schein, J., Birol, I., Connors, J., Gascoyne, R., Horsman, D., Jones, S.J., and Marra, M.A. (2009). Circos: an information aesthetic for comparative genomics. Genome Res 19, 1639-1645.
