By conducting pairwise genome alignment between each de novo assembled rhinoceros genome and that of horse (Equus caballus), we did not detect any major within-scaffold segment reshuffling between the rhinoceros species, as well as for that between rhinoceroses and the horse. We then applied Circos (Krzywinski et al., 2009) to visualise the genome synteny and collinearity (Figure 1). The good within-scaffold collinearity between rhinoceros species to some extent guarantees the reliability and accuracy of the reference-based analyses for those historic and ancient samples.


![Figure 1](https://github.com/liushanlin/rhinoceros-comparative-genome/blob/main/additional%20resources/Equu_chr1.png)

**Figure 1. A demonstration example shows the pairwise genome alignments between the de novo-assembled genomes and horse using chromosome #1**. Species can be identified by the first letter of each scaffold ID: “w” represents white rhinoceros (red lines), “b” represents black rhinoceros (blue lines), “s” represents Sumatran rhinoceros (green lines) and “i” represents greater one-horned rhinoceros (orange lines). The figure only shows scaffolds of length > 5 Mb and alignment blocks of length > 100 kb.

> **Reference**
> 
> Krzywinski, M., Schein, J., Birol, I., Connors, J., Gascoyne, R., Horsman, D., Jones, S.J., and Marra, M.A. (2009). Circos: an information aesthetic for comparative genomics. Genome Res 19, 1639-1645.
